tests/github_app_test.py

```diff
@@ -0,0 +1,72 @@
+from typing import cast
+from unittest import mock
+
+import pytest
+
+from github import Auth
+
+from awx_plugins.credentials.github_app import github_app_backend
+
+
+JWT_EXPIRY_DEFAULT = 600
+
+
+def test_github_app_missing_parameters() -> None:
+    """Test that missing parameters raise an exception."""
+    with pytest.raises(ValueError, match=r'Missing Parameter: \[.*\]'):
+        github_app_backend()
+
+    with pytest.raises(ValueError, match=r"Missing Parameter: \['GitHub URL'\]"):
+        github_app_backend(
+            # type: ignore[arg-type]
+            app_id='123', install_id='456', ssh_key_data='key',
+        )
+
+
+def test_github_app_invalid_app_id_and_install_id() -> None:
+    """Test that non-integer app_id and install_id raise an exception."""
+    with pytest.raises(ValueError, match=r'App ID and Installation ID must be integers .*invalid literal for int\(\) with base 10: .*'):
+        github_app_backend(
+            github_url='https://api.github.com',  # type: ignore[arg-type]
+            app_id='invalid',  # type: ignore[arg-type]
+            install_id='invalid',  # type: ignore[arg-type]
+            ssh_key_data='key',  # type: ignore[arg-type]
+        )
+
+
+def test_github_app_invalid_jwt_expiry() -> None:
+    """Test that non-integer JWT expiry raises an exception."""
+    with pytest.raises(ValueError, match=r'JWT Expiry must be an integer invalid'):
+
+        github_app_backend(
+            github_url='https://api.github.com',  # type: ignore[arg-type]
+            app_id='123',  # type: ignore[arg-type]
+            install_id='456',  # type: ignore[arg-type]
+            ssh_key_data='key',  # type: ignore[arg-type]
+            jwt_expiry='invalid',  # type: ignore[arg-type]
+        )
```

Comment:
These exercise the same code path and are mostly identical. They can be merged into a single test function and parametrized.